fsm/fsm_test.go

```diff
@@ -1,128 +1,111 @@
-	fsmConfig, err := fsm.LoadConfig(testutils.Examples04TriviaPath)
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	domain := fsm.New(fsmConfig)
-
-	machine := fsm.FSM{
-		State: 1,
-		Slots: make(map[string]string),
-	}
-
-	machine.ExecuteCmd("start", "1", domain)
-}
-
-func TestCacheStore(t *testing.T) {
-	machines := fsm.NewStore(fsm.StoreConfig{Type: "CACHE"})
-
-	if resp1 := machines.Exists("foo"); resp1 != false {
-		t.Errorf("incorrect, got: %v, want: %v.", resp1, "false")
-	}
-
-	machines.Set(
-		"foo",
-		&fsm.FSM{
-			State: 0,
-			Slots: make(map[string]string),
+var domainOnOff = fsm.NewDomain(
+	[]string{"turn_on", "turn_off"},
+	[]string{"off", "on"},
+	[]fsm.Function{
+		{
+			Transition: fsm.Transition{
+				From: "off",
+				Into: "on",
+			},
+			Command: "turn_on",
+			Message: []fsm.Message{{
+				Text: "Turning on.",
+			}},
 		},
-	)
-	if resp2 := machines.Get("foo"); resp2.State != 0 {
-		t.Errorf("incorrect, got: %v, want: %v.", resp2, "0")
-	}
-
-	newFsm := &fsm.FSM{
-		State: 1,
-		Slots: map[string]string{
-			"abc": "xyz",
+		{
+			Transition: fsm.Transition{
+				From: "on",
+				Into: "off",
+			},
+			Command: "turn_off",
+			Message: []fsm.Message{
+				{
+					Text: "Turning off.",
+				},
+				{
+					Text: "❌",
+				},
+			},
 		},
-	}
-	machines.Set("foo", newFsm)
-	if resp3 := machines.Get("foo"); resp3.State != 1 {
-		t.Errorf("incorrect, got: %v, want: %v.", resp3, "1")
-	}
-}
-
-func TestRedisStore(t *testing.T) {
-	machines := fsm.NewStore(fsm.StoreConfig{
-		Type:     "REDIS",
-		Host:     "localhost",
-		Password: "pass",
-	})
+	},
+	fsm.Defaults{
+		Unknown: "Can't do that.",
+		Unsure:  "???",
+		Error:   "Error",
+	},
+)
 
-	if resp1 := machines.Exists("foo"); resp1 != false {
-		t.Errorf("incorrect, got: %v, want: %v.", resp1, "false")
+func TestFSM_ExecuteCmd(t *testing.T) {
+	type fields struct {
+		State int
+		Slots map[string]string
 	}
-
-	machines.Set(
-		"foo",
-		&fsm.FSM{
-			State: 0,
-			Slots: make(map[string]string),
-		},
-	)
-	if resp2 := machines.Get("foo"); resp2.State != 0 {
-		t.Errorf("incorrect, got: %v, want: %v.", resp2, "0")
+	type args struct {
+		cmd       string
+		txt       string
+		fsmDomain *fsm.Domain
 	}
-
-	newFsm := &fsm.FSM{
-		State: 1,
-		Slots: map[string]string{
-			"abc": "xyz",
+	tests := []struct {
+		name          string
+		fields        fields
+		args          args
+		wantAnswers   []query.Answer
+		wantExtension string
+	}{
+		{
+			fields: fields{
+				State: 0,
+				Slots: make(map[string]string),
+			},
+			args: args{
+				cmd:       "turn_on",
+				txt:       "blarg",
+				fsmDomain: domainOnOff,
+			},
+			wantAnswers: []query.Answer{{
+				Text: "Can't do that.",
+			}},
+			wantExtension: "",
+		},
+		{
+			fields: fields{
+				State: 0,
+				Slots: make(map[string]string),
+			},
+			args: args{
+				cmd:       "turn_on",
+				txt:       "on",
+				fsmDomain: domainOnOff,
+			},
+			wantAnswers: []query.Answer{{
+				Text: "Turning on.",
```

Comment:
I cannot get this test case to return "Turning on.". I'm not sure what I'm doing wrong here. I'm digging into this currently. Trying to understand how it works.